packages/backend/src/database/entities/savedQueries.ts

```diff
@@ -0,0 +1,287 @@
+                descending: sort.descending,
+            })),
+            limit: savedQuery.row_limit,
+        },
+        chartConfig: {},
+    };
+};
+
+const createSavedQueryVersionField = async (
+    db: Knex,
+    data: CreateDbSavedQueryVersionField,
+): Promise<DbSavedQueryVersionField> => {
+    const results = await db<DbSavedQueryVersionField>(
+        'saved_queries_version_fields',
+    )
+        .insert<CreateDbSavedQueryVersionField>(data)
+        .returning('*');
+    return results[0];
+};
+
+const createSavedQueryVersionSort = async (
+    db: Knex,
+    data: CreateDbSavedQueryVersionSort,
+): Promise<DbSavedQueryVersionSort> => {
+    const results = await db<DbSavedQueryVersionSort>(
+        'saved_queries_version_sorts',
+    )
+        .insert<CreateDbSavedQueryVersionSort>(data)
+        .returning('*');
+    return results[0];
+};
+
+export const createSavedQueryVersion = async (
+    db: Knex,
+    savedQueryId: number,
+    {
+        tableName,
+        metricQuery: { limit, filters, dimensions, metrics, sorts },
+    }: CreateSavedQueryVersion,
+): Promise<void> => {
+    await db.transaction(async (trx) => {
+        try {
+            const results = await trx<DbSavedQueryVersion>(
+                'saved_queries_versions',
+            )
+                .insert<CreateDbSavedQueryVersion>({
+                    row_limit: limit,
+                    filters: JSON.stringify(filters),
+                    explore_name: tableName,
+                    saved_query_id: savedQueryId,
+                })
+                .returning('*');
+            const version = results[0];
+
+            const promises: Promise<any>[] = [];
+            for (let index = 0; index < dimensions.length; index += 1) {
+                promises.push(
+                    createSavedQueryVersionField(trx, {
+                        name: dimensions[index],
+                        isDimension: true,
+                        saved_queries_version_id:
+                            version.saved_queries_version_id,
+                    }),
+                );
+            }
+            for (let index = 0; index < metrics.length; index += 1) {
+                promises.push(
+                    createSavedQueryVersionField(trx, {
+                        name: metrics[index],
+                        isDimension: false,
+                        saved_queries_version_id:
+                            version.saved_queries_version_id,
+                    }),
+                );
+            }
+            for (let index = 0; index < sorts.length; index += 1) {
+                promises.push(
+                    createSavedQueryVersionSort(trx, {
+                        field_name: sorts[index].fieldId,
+                        descending: sorts[index].descending,
+                        saved_queries_version_id:
+                            version.saved_queries_version_id,
+                    }),
+                );
+            }
```

Comment:
I don't think this guarantees insert order in any way, is that important if you want to rely on the incrementing key to determine order later?